src/flint/flint_base/flint_base.pyx

```diff
@@ -869,6 +941,81 @@ cdef class flint_mpoly(flint_elem):
         """
         return zip(self.monoms(), self.coeffs())
 
+    def unused_gens(self):
+        """
+        Report the unused generators from this polynomial.
+
+        A generator is unused if it's maximum degree is 0.
+
+            >>> from flint import fmpz_mpoly_ctx
+            >>> ctx = fmpz_mpoly_ctx.get(('x', 4))
+            >>> ctx2 = fmpz_mpoly_ctx.get(('x1', 'x2'))
+            >>> f = sum(ctx.gens()[1:3])
+            >>> f
+            x1 + x2
+            >>> f.unused_gens()
+            ('x0', 'x3')
+        """
+        names = self.context().names()
+        return tuple(names[i] for i, x in enumerate(self.degrees()) if not x)
+
+    def project_to_context(self, other_ctx, mapping: dict[str | int, str | int] = None):
+        """
+        Project this polynomial to a different context.
+
+        This is equivalent to composing this polynomial with the generators of another
+        context. By default the mapping between contexts is inferred based on the name
+        of the generators. Generators with names that are not found within the other
+        context are mapped to 0. The mapping can be explicitly provided.
+
+            >>> from flint import fmpz_mpoly_ctx
+            >>> ctx = fmpz_mpoly_ctx.get(('x', 'y', 'a', 'b'))
+            >>> ctx2 = fmpz_mpoly_ctx.get(('a', 'b'))
+            >>> x, y, a, b = ctx.gens()
+            >>> f = x + 2*y + 3*a + 4*b
+            >>> f.project_to_context(ctx2)
+            3*a + 4*b
+            >>> f.project_to_context(ctx2, mapping={"x": "a", "b": 0})
+            5*a
```

Comment:
Setting `"b": 0` here seems ambiguous to me. Is it setting "b" to the variable of index 0 or to the value 0?

Reply:
Fair. It means the generator at index 0 here. I meant to show that it also accepts generator indices. Perhaps I should make that clearer